### ISSUE TITLE
1.1.2 and 1.1.1 multiple definition of `aesni_encrypt'

### ISSUE THREAD

**Issue by fengshh2301** [pokerth/pokerth#345]
I want to compile pokerth under mxe, but got these msg:

/opt/mingw/usr/i686-w64-mingw32.static/lib/libcrypto.a(aesni-x86.o):crypto/aes/asm/aesni-x86.s:(.text+0x0): multiple definition of `aesni_encrypt'
/opt/mingw/usr/i686-w64-mingw32.static/lib/libgnutls.a(aesni-x86.o):/opt/mingw/tmp-gnutls-i686-w64-mingw32.static/gnutls-3.5.18/lib/accelerated/x86/coff/aesni-x86.s:47: first defined here
/opt/mingw/usr/i686-w64-mingw32.static/lib/libcrypto.a(aesni-x86.o):crypto/aes/asm/aesni-x86.s:(.text+0x50): multiple definition of `aesni_decrypt'
/opt/mingw/usr/i686-w64-mingw32.static/lib/libgnutls.a(aesni-x86.o):/opt/mingw/tmp-gnutls-i686-w64-mingw32.static/gnutls-3.5.18/lib/accelerated/x86/coff/aesni-x86.s:70: first defined here
/opt/mingw/usr/i686-w64-mingw32.static/lib/libcrypto.a(aesni-x86.o):crypto/aes/asm/aesni-x86.s:(.text+0x500): multiple definition of `aesni_ecb_encrypt'
/opt/mingw/usr/i686-w64-mingw32.static/lib/libgnutls.a(aesni-x86.o):/opt/mingw/tmp-gnutls-i686-w64-mingw32.static/gnutls-3.5.18/lib/accelerated/x86/coff/aesni-x86.s:335: first defined here
/opt/mingw/usr/i686-w64-mingw32.static/lib/libcrypto.a(aesni-x86.o):crypto/aes/asm/aesni-x86.s:(.text+0x840): multiple definition of `aesni_ccm64_encrypt_blocks'
/opt/mingw/usr/i686-w64-mingw32.static/lib/libgnutls.a(aesni-x86.o):/opt/mingw/tmp-gnutls-i686-w64-mingw32.static/gnutls-3.5.18/lib/accelerated/x86/coff/aesni-x86.s:563: first defined here
/opt/mingw/usr/i686-w64-mingw32.static/lib/libcrypto.a(aesni-x86.o):crypto/aes/asm/aesni-x86.s:(.text+0x970): multiple definition of `aesni_ccm64_decrypt_blocks'
/opt/mingw/usr/i686-w64-mingw32.static/lib/libgnutls.a(aesni-x86.o):/opt/mingw/tmp-gnutls-i686-w64-mingw32.static/gnutls-3.5.18/lib/accelerated/x86/coff/aesni-x86.s:642: first defined here
/opt/mingw/usr/i686-w64-mingw32.static/lib/libcrypto.a(aesni-x86.o):crypto/aes/asm/aesni-x86.s:(.text+0xb10): multiple definition of `aesni_ctr32_encrypt_blocks'
/opt/mingw/usr/i686-w64-mingw32.static/lib/libgnutls.a(aesni-x86.o):/opt/mingw/tmp-gnutls-i686-w64-mingw32.static/gnutls-3.5.18/lib/accelerated/x86/coff/aesni-x86.s:753: first defined here
/opt/mingw/usr/i686-w64-mingw32.static/lib/libcrypto.a(aesni-x86.o):crypto/aes/asm/aesni-x86.s:(.text+0xea0): multiple definition of `aesni_xts_encrypt'
/opt/mingw/usr/i686-w64-mingw32.static/lib/libgnutls.a(aesni-x86.o):/opt/mingw/tmp-gnutls-i686-w64-mingw32.static/gnutls-3.5.18/lib/accelerated/x86/coff/aesni-x86.s:983: first defined here
/opt/mingw/usr/i686-w64-mingw32.static/lib/libcrypto.a(aesni-x86.o):crypto/aes/asm/aesni-x86.s:(.text+0x1410): multiple definition of `aesni_xts_decrypt'
/opt/mingw/usr/i686-w64-mingw32.static/lib/libgnutls.a(aesni-x86.o):/opt/mingw/tmp-gnutls-i686-w64-mingw32.static/gnutls-3.5.18/lib/accelerated/x86/coff/aesni-x86.s:1329: first defined here
/opt/mingw/usr/i686-w64-mingw32.static/lib/libcrypto.a(aesni-x86.o):crypto/aes/asm/aesni-x86.s:(.text+0x26d0): multiple definition of `aesni_cbc_encrypt'
/opt/mingw/usr/i686-w64-mingw32.static/lib/libgnutls.a(aesni-x86.o):/opt/mingw/tmp-gnutls-i686-w64-mingw32.static/gnutls-3.5.18/lib/accelerated/x86/coff/aesni-x86.s:1704: first defined here
/opt/mingw/usr/i686-w64-mingw32.static/lib/libcrypto.a(aesni-x86.o):crypto/aes/asm/aesni-x86.s:(.text+0x2f40): multiple definition of `aesni_set_encrypt_key'
/opt/mingw/usr/i686-w64-mingw32.static/lib/libgnutls.a(aesni-x86.o):/opt/mingw/tmp-gnutls-i686-w64-mingw32.static/gnutls-3.5.18/lib/accelerated/x86/coff/aesni-x86.s:2119: first defined here
/opt/mingw/usr/i686-w64-mingw32.static/lib/libcrypto.a(aesni-x86.o):crypto/aes/asm/aesni-x86.s:(.text+0x2f60): multiple definition of `aesni_set_decrypt_key'
/opt/mingw/usr/i686-w64-mingw32.static/lib/libgnutls.a(aesni-x86.o):/opt/mingw/tmp-gnutls-i686-w64-mingw32.static/gnutls-3.5.18/lib/accelerated/x86/coff/aesni-x86.s:2129: first defined here
collect2: error: ld returned 1 exit status
make[2]: *** [release/pokerth.exe] Error 1
make[2]: Leaving directory `/home/jason/workspace/pokerth'
make[1]: *** [release] Error 2
make[1]: Leaving directory `/home/jason/workspace/pokerth'
make: *** [sub-pokerth_game-pro-release_ordered] Error 2

i think there is some conflict between crypto and gnutls, so how can i fix this issue?

**Comment by albmed** [comment]
Hi @fengshh2301 

Did you take a look at [Building PokerTH for Windows wiki](https://github.com/pokerth/pokerth/wiki/Building-PokerTH-for-Windows)? 

I think I can't help. It looks, it has nothing to do with PokerTH, but with your own MXE libraries. 

Is `pokerth.exe` created in `./release` directory? 
Sometimes MXE throws some errors during linking process, but it builds PokerTH, anyway. This kind of errors are usually because of MXE itself.  

Regards.

**Comment by fengshh2301** [comment]
Ye, i am sure i keep steps after 'Building PokerTH for Windows wiki'.
In fact, I do all same thing to the wiki.
Can not understand why the 'aesni_encrypt' exists in two libs: libcrypto.a and libgnutls.a

**Comment by albmed** [comment]
Yes, me neither... We used to have a similar problem with `protoc` library when we built windows versions, but `pokerth.exe` was created, anyway. So, we just supposed that those were MXE errors and we ignored them. That's why the [Known bugs section](https://github.com/pokerth/pokerth/wiki/Building-PokerTH-for-Windows#known-bugs). 

On the other hand, this looks to be a linker error, since it refers to object files and not to source files. If `./release/pokerth.exe` was not created, check if there is a compilation error before that liker error in a `cpp` file, such as:  

```
call of overloaded ‘aesni_encrypt’ is ambiguous
```

It could mean there is a missing namespace qualifying in source, because of a new version of any of those libraries. In that case, just tell us, please. 

Regards.

**Comment by q4z1** [comment]
Outdated.
